runtime/lib/mirrors.cc

```diff
@@ -835,12 +835,59 @@ DEFINE_NATIVE_ENTRY(Mirrors_makeLocalClassMirror, 1) {
                            Object::null_instance());
 }
 
-
 DEFINE_NATIVE_ENTRY(Mirrors_makeLocalTypeMirror, 1) {
   GET_NON_NULL_NATIVE_ARGUMENT(AbstractType, type, arguments->NativeArgAt(0));
   return CreateTypeMirror(type);
 }
 
+DEFINE_NATIVE_ENTRY(Mirrors_makeLocalTypeMirrorWithTypeArguments, 2) {
+  GET_NON_NULL_NATIVE_ARGUMENT(AbstractType, type, arguments->NativeArgAt(0));
+  GET_NON_NULL_NATIVE_ARGUMENT(Array, args, arguments->NativeArgAt(1));
+
+  const Class& clz = Class::Handle(type.type_class());
```

Comment:
I'm assuming 1st and 3rd lines should throw an error and the 2nd line should return mirror of `Predicate<int>`?

I'm working on it and it works almost as expected, with couple issues:

* I can't really construct `Predicate<int>` as I do with normal classes using `new A().runtimeType`. So I haven't figured yet how to write this assertion.
* Looks like bounded types are not being checked atm. I'm figuring out how to fix this.